packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/heatmap.dart

```diff
@@ -0,0 +1,371 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart'
+    show immutable, listEquals, objectRuntimeType;
+import 'package:flutter/material.dart' show Color;
+
+import 'types.dart';
+
+/// Uniquely identifies a [Heatmap] among [GoogleMap] heatmaps.
+///
+/// This does not have to be globally unique, only unique among the list.
+@immutable
+class HeatmapId extends MapsObjectId<Heatmap> {
+  /// Creates an immutable identifier for a [Heatmap].
+  const HeatmapId(super.value);
+}
+
+/// A wrapper around platform specific behavior for the radius of a [Heatmap].
+///
+/// Currently this class only handles platform-specific values, but in the
+/// future it may provide alternate constructors that abstract platform
+/// differences in handling of heatmap radius values.
+///
+/// See https://github.com/flutter/flutter/issues/145411
+///
+/// TODO(stuartmorgan): Add constructor and enum field that informs the platform how to perform the conversion.
+@immutable
+class HeatmapRadius {
```

Comment:
@reidbaker 
I closed all other discussions about the `HeatmapRadius` class since they're buried above. @stuartmorgan and I discussed this offline and decided this is the best solution for now. In the future, we will add an enum field that informs the platform how to perform the conversion, or to not perform any conversion if `HeatmapRadius.fromPixels` was used.